cloudbuild.yaml

```diff
@@ -22,6 +23,41 @@ steps:
   - run
   - '--rm'
   - snuba-test
+# Clean up after tests
+- name: 'gcr.io/$PROJECT_ID/docker-compose'
+  env:
+  - 'SNUBA_IMAGE=us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA'
+  args:
+  - '-f'
+  - 'docker-compose.gcb.yml'
+  - down
+  - '--rmi'
+  - 'local'
+  - '-v'
+  - '--remove-orphans'
+# On-premise Integration tests
+- name: 'gcr.io/$PROJECT_ID/docker-compose'
+  entrypoint: 'bash'
+  env:
+  - 'SNUBA_IMAGE=us.gcr.io/$PROJECT_ID/snuba:$COMMIT_SHA'
+  - 'SENTRY_TEST_HOST=http://nginx'
+  - 'CI=1'
+  args:
+  - '-e'
+  - '-c'
+  - |
+    mkdir onpremise && cd onpremise
+    # TODO(byk): We may also build this part as a builder image everytime
+    #            there's a push to onpremise and use that image for
+    curl -L "https://github.com/getsentry/onpremise/archive/master.tar.gz" | tar xzf - --strip-components=1
+    # The following trick is from https://stackoverflow.com/a/52400857/90297 with great gratuity
+    echo '{"version": "3.4", "networks":{"default":{"external":{"name":"cloudbuild"}}}}' > docker-compose.override.yml
+    ./install.sh
+    docker-compose run --rm web createuser --superuser --email test@example.com --password test123TEST
+    docker-compose up -d
+    timeout 20 bash -c 'until $(curl -Isf -o /dev/null http://nginx); do printf "."; sleep 0.5; done' || docker-compose logs web
+    ./test.sh || docker-compose logs nginx web relay
```

Comment:
Which kinds of integration tests do you expect going into this? Asking because this makes Snuba cloudbuild 2-3 minutes longer (it was 5-6 minutes  now it it 8). Do we expect test.sh to ever include all Sentry tests or things like that ?

Reply:
> Do we expect test.sh to ever include all Sentry tests or things like that ?

Definitely not. This repo tests the basic functionality in a fully-connected service environment. The most we'll do is adding a few more basic curl-based tests like this for Symbolication and performance at the moment.